### PR TITLE
fix: when updating rows with a ref array, if the list is null, replace with an empty array

### PR DESCRIPTION
**Pull request by jhhaanstra**
Addresses https://github.com/molgenis/GCC/issues/2821

### What are the main changes you did
- Null and empty lists are both mapped to null in the database for ref_arrays

### How to test
- unit test
- Execute reproduce steps in issue: https://github.com/molgenis/GCC/issues/2821

### Checklist
- [ ] updated docs in case of new feature
- [X] added/updated tests
- [X] added/updated testplan to include a test for this fix, including ref to bug using # notation

By creating this pull request I have agreed with the [contributor terms](https://github.com/molgenis/molgenis-emx2/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/utils/TypeUtils.java [hunk 1]
```diff
@@ -551,11 +551,12 @@ public static void addFieldObjectToRow(Column column, Object object, Row row) {
 
   protected static void convertRefArrayToRow(
       List<Map<String, Object>> list, Row row, Column column) {
-    if (list == null) return;
     List<Reference> refs = column.getReferences();
     for (Reference ref : refs) {
       if (!ref.isOverlapping()) {
-        if (!list.isEmpty()) {
+        if (list == null) {
```
Comment (harmbrugge) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/utils/TypeUtils.java]: You can merge == null and isEmpty right? bit more readable:

```
if (list == null || list.isEmpty()) {
  row.set(ref.getName(), new ArrayList<>());
} else {
  row.set(ref.getName(), getRefValueFromList(ref.getPath(), list));
}
```

**Review by harmbrugge — CHANGES_REQUESTED** [review]

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/utils/TypeUtils.java (hunk 1 above)
Comment (jhhaanstra) [reply backend/molgenis-emx2/src/main/java/org/molgenis/emx2/utils/TypeUtils.java]: Ah yes, thanks! 🙏

**Review by MaxPostema — APPROVED** [review]
Works, cant find issues. Thanks! 👍

**Review by harmbrugge — APPROVED** [review]
Code looks good. Not entirely sure if this desired behaviour

**Review by dtroelofsprins — APPROVED** [review]
Update via graphQL works fine, both with null as well as []. Also tested an ontology array, checkbox and multiselect. All work fine.

Also tested file upload and this still also works fine (except that you cannot enter empty list there, but null always did work fine and still is working fine).

NB: In the related issue it was stated that when currently using null value, it would say that 1 row has been updated, but actually the value was not updated. This however didn't was when I was testing it. It said 0 rows updated.

File: backend/molgenis-emx2/src/main/java/org/molgenis/emx2/utils/TypeUtils.java [hunk 2]
```diff
@@ -551,14 +551,13 @@ public static void addFieldObjectToRow(Column column, Object object, Row row) {
 
   protected static void convertRefArrayToRow(
       List<Map<String, Object>> list, Row row, Column column) {
-    if (list == null) return;
     List<Reference> refs = column.getReferences();
     for (Reference ref : refs) {
       if (!ref.isOverlapping()) {
-        if (!list.isEmpty()) {
-          row.set(ref.getName(), getRefValueFromList(ref.getPath(), list));
+        if (list == null || list.isEmpty()) {
```
Comment (mswertz) [inline backend/molgenis-emx2/src/main/java/org/molgenis/emx2/utils/TypeUtils.java:557]: yes, this looks right

**Review by mswertz — APPROVED** [review]
